astrokat/noisediode.py

```diff
@@ -9,257 +9,434 @@
+        time.sleep(sleeptime)
+        user_logger.trace('TRACE: ts after sleep {} ({})'
+                          .format(time.time(),
+                                  time.ctime(time.time())))
+    else:
+        user_logger.trace('TRACE: Trigger duration <= lead_time')
+        cycle_len = _get_max_cycle_len(kat)
+        nd_setup = {'antennas': 'all',
+                    'cycle_len': cycle_len,
+                    'on_frac': float(duration) / cycle_len,
+                    }
+        user_logger.debug('DEBUG: fire nd for {} using pattern'
+                          .format(duration))
+        on_time = pattern(kat, nd_setup, lead_time=lead_time)
+        user_logger.debug('DEBUG: pattern set {} ({})'
+                          .format(on_time,
+                                  time.ctime(on_time)))
+        off_time = _get_nd_timestamp_(lead_time)
+        user_logger.trace('TRACE: desired off_time {} ({})'
+                          .format(off_time,
+                                  time.ctime(off_time)))
+
+    user_logger.debug('DEBUG: off {} ({})'
+                      .format(off_time,
+                              time.ctime(off_time)))
+    off_time = off(kat, timestamp=off_time)
+    sleeptime = off_time - time.time()
+    user_logger.debug('DEBUG: now {}, sleep {}'
+                      .format(time.time(),
+                              sleeptime))
     time.sleep(sleeptime)  # default sleep to see for signal to get through
-    user_logger.debug("DEBUG: now {}, slept {}".format(time.time(), sleeptime))
-    return True
+    user_logger.debug('DEBUG: now {}, slept {}'
+                      .format(time.time(),
+                              sleeptime))
 
 
-def pattern(kat, session, nd_setup, lead_time=_DEFAULT_LEAD_TIME):
+# set noise diode pattern
+def pattern(kat,
+            nd_setup,
+            lead_time=_DEFAULT_LEAD_TIME,
+            ):
     """Start background noise diode pattern controlled by digitiser hardware.
 
     Parameters
     ----------
-    kat : Session kat container-like object
+    kat : session kat container-like object
         Container for accessing KATCP resources allocated to schedule block.
-    session : katcorelib.CaptureSession-like object
-    nd_setup: dict
+    nd_setup : dict
         Noise diode pattern setup, with keys:
-            'antennas':  options are `ants`, or 'm062',...
-            'cycle_len': the cycle length [sec], - must be less than 20 sec for L-band
-    lead_time : float, optional
-                Lead time before digitisers pattern is set [sec]
-
+            'antennas':  options are 'all', or 'm062', or ....,
+            'cycle_len': the cycle length [sec],
+                           - must be less than 20 sec for L-band,
+            etc., etc.
+    lead_time : float, optional (default = system default lead time)
+        Lead time before digitisers pattern is set [sec]
+
+    Returns
+    -------
+    timestamp: float
```

Comment:
For consistency perhaps add a space before the colon.
```suggestion
    timestamp : float
```

Reply:
corrected docstring format